app/models/authenticator.rb

```diff
@@ -34,9 +34,23 @@ def uses_stored_password?
       false
     end
 
+    def authorize_user_by_userid?
```

Comment:
naming is hard, was going by other method names in this file, i.e. authorize_queue?, no worry, I can change this.